elki/src/main/java/ch/ethz/globis/pht/pre/ColumnType.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2011-2015 ETH Zurich. All Rights Reserved.
+ *
+ * This software is the proprietary information of ETH Zurich.
+ * Use is subject to license terms.
+ */
+package ch.ethz.globis.pht.pre;
+
+public abstract class ColumnType {
```

Comment:
JavaDoc for this class would be good, in case someone wants to extend it to new column types. I know it isn't fun, and `depth` is not hard to understand.